padl/transforms.py

```diff
@@ -639,12 +641,15 @@ def pd_forward_device_check(self) -> bool:
         All transforms in forward need to be in same device as specified for
         the whole Pipeline.
         """
-        for layer in self.pd_forward.pd_layers:
+        pd_device = self.pd_device
+        pd_layers = self.pd_forward.pd_layers
+        for layer in pd_layers:
             for parameters in layer.parameters():
                 parameter_device = parameters.device.type
-                if ':' in self.pd_device and 'cuda' in parameter_device:
-                    parameter_device += f':{parameters.device.index}'
-                if parameter_device != self.pd_device:
+                parameter_device_index = parameters.device.index
```

Comment:
Same here.

Reply:
changing back

Reply:
You resolved, but it's still the same isn't it? Point was that as parameter_device_index is only used once we don't need to assign it to a variable.

Reply:
I believe I changed this back. The code snippet is shown to be out of date at the top of this comment.